middlewares/replace_path.go

```diff
@@ -0,0 +1,21 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+// ReplacePath is a middleware used to replace the path of a URL request
+type ReplacePath struct {
+	Handler http.Handler
+	Path    string
+}
+
+// ReplacedPathHeader is the default header to set the old path to
+const ReplacedPathHeader = "X-Replaced-Path"
+
+func (s *ReplacePath) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.Header.Add(ReplacedPathHeader, r.URL.Path)
+	r.URL.Path = s.Path
+	r.RequestURI = r.URL.RequestURI()
```

Comment:
I did find it a bit strange too, but iirc, it does not directly affect a client request.

I'm probably wrong, but at one point there might have been some code that read r.RequestURI` to use for the path to use for the backend request. That would also explain why the same expression elsewhere in the code base.